hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java

```diff
@@ -146,45 +163,43 @@ private void established(Channel ch) throws IOException {
   private boolean reloginInProgress;
 
   private void scheduleRelogin(Throwable error) {
+    assert eventLoop.inEventLoop();
```

Comment:
just checking you want these to be java asserts? production jdks are almost never run with asserts enabled

Reply:
My intention is to let later developpers know that this method must be called in the event loop thread. Do not want to introduce any overhead when executing the code in production, as in production, if it is not called in an event loop thread, it will be a ctitical bug and can not be recovered, so I do not think it worth other checks like Preconditions.checkXXX...